server/graphql/invoice/src/mutations/inbound_shipment/add_from_master_list.rs

```diff
@@ -0,0 +1,311 @@
+        let test_service = TestService(Box::new(|_, _| Err(ServiceError::ShipmentDoesNotExist)));
+
+        let expected = json!({
+            "addToInboundShipmentFromMasterList": {
+              "error": {
+                "__typename": "RecordNotFound"
+              }
+            }
+          }
+        );
+
+        assert_graphql_query!(
+            &settings,
+            mutation,
+            &Some(empty_variables()),
+            &expected,
+            Some(service_provider(test_service, &connection_manager))
+        );
+
+        // CannotEditInvoice
+        let test_service = TestService(Box::new(|_, _| Err(ServiceError::CannotEditShipment)));
+
+        let expected = json!({
+            "addToInboundShipmentFromMasterList": {
+              "error": {
+                "__typename": "CannotEditInvoice"
+              }
+            }
+          }
+        );
+
+        assert_graphql_query!(
+            &settings,
+            mutation,
+            &Some(empty_variables()),
+            &expected,
+            Some(service_provider(test_service, &connection_manager))
+        );
+
+        // MasterListNotFoundForThisName
+        let test_service = TestService(Box::new(|_, _| {
+            Err(ServiceError::MasterListNotFoundForThisName)
+        }));
+
+        let expected = json!({
+            "addToInboundShipmentFromMasterList": {
+              "error": {
+                "__typename": "MasterListNotFoundForThisName"
+              }
+            }
+          }
+        );
+
+        assert_graphql_query!(
+            &settings,
+            mutation,
+            &Some(empty_variables()),
+            &expected,
+            Some(service_provider(test_service, &connection_manager))
+        );
+
+        // NotThisStoreInvoice
+        let test_service = TestService(Box::new(|_, _| Err(ServiceError::NotThisStoreShipment)));
+        let expected_message = "Bad user input";
+        assert_standard_graphql_error!(
+            &settings,
+            &mutation,
+            &Some(empty_variables()),
+            &expected_message,
+            None,
+            Some(service_provider(test_service, &connection_manager))
+        );
+
+        // NotAnInboundShipment
+        let test_service = TestService(Box::new(|_, _| Err(ServiceError::NotAnInboundShipment)));
+        let expected_message = "Bad user input";
+        assert_standard_graphql_error!(
+            &settings,
+            &mutation,
+            &Some(empty_variables()),
+            &expected_message,
+            None,
+            Some(service_provider(test_service, &connection_manager))
+        );
+    }
+
+    #[actix_rt::test]
+    async fn test_graphql_add_from_master_list_success() {
+        let (_, _, connection_manager, settings) = setup_graphl_test(
+            EmptyMutation,
+            InvoiceMutations,
+            "test_graphql_add_is_from_master_list_success",
+            MockDataInserts::all(),
```

Comment:
Would be good to use MokDataInserts::none(), i don't think we use any data in this test

Reply:
The test is using data - I tried specifying just the required bits, but even then had foreign key errors, so gave up. have put `all()` back in.

Reply:
it worked for me with `none()`. Graphql tests are just using service overrides to check mapping between service and graphql, they shouldn't be using any mock data inserted in database. The reason we need connection_manager here is to instantiate test graphql instance.

Reply:
huh. worked that time :shrug:

Reply:
randomly I had the thought to do something like this yesterday:

```rust
impl MockDataInserts {
    pub fn stores(mut self) -> Self {
        self.names = true;
        self.stores = true;
        self
    }
```